src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +22,31 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    private static final int LETTERS_AMOUNT = 5;
+
+    public String getUniqueCharacters(String fileName) throws IOException {
+        TreeSet<Character> lettersFromFile = new TreeSet<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            int input = reader.read();
+            while (input != -1) {
+                if (input >= 'A' && input <= 'Z') {
+                    input += 32;
+                }
+                if (input >= 'a' && input <= 'z') {
+                    lettersFromFile.add((char) input);
+                }
+                input = reader.read();
+            }
+        }
+        StringBuilder result = new StringBuilder();
+        int border = LETTERS_AMOUNT;
+        for (char letter : lettersFromFile) {
+            result.append(letter);
+            if (border == 1) {
```

Comment:
suggestion: 
`if (result.length == LETTERS_AMOUNT) {
  break;
}`

Reply:
then you can get rid of border variable.

Reply:
thank you, thats truly better